### PR TITLE
Condition set object type

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
#2571
Author Checklist

Changes address original issue? | Y
Unit tests included and/or updated with changes? | Y
Command line build passes? | Y
Changes have been smoke-tested? | Y

### CODE REVIEW

File: src/MCT.js [hunk 1]
```diff
@@ -264,6 +264,7 @@ define([
         this.install(this.plugins.GoToOriginalAction());
         this.install(this.plugins.ImportExport());
         this.install(this.plugins.WebPage());
+        this.install(this.plugins.Condition());
```
Comment (shefalijoshi) [inline src/MCT.js]: Shouldn't this be ConditionSet?

File: src/plugins/condition/plugin.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export default function ConditionSetPlugin() {
+    const conditionSetType = {
+        name: 'Condition Set',
+        key: 'conditionSet',
+        description: 'A set of one or more conditional rules based on user-specified criteria.',
+        creatable: true,
+        cssClass: 'icon-summary-widget',
+        initialize: function (domainObject) {
+            domainObject.composition = [];
+            domainObject.telemetry = {};
```
Comment (shefalijoshi) [inline src/plugins/condition/plugin.js]: I don't think we need this telemetry object. Let's remove it for now.

File: src/plugins/condition/plugin.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+export default function ConditionSetPlugin() {
+    const conditionSetType = {
+        name: 'Condition Set',
+        key: 'conditionSet',
+        description: 'A set of one or more conditional rules based on user-specified criteria.',
```
Comment (shefalijoshi) [inline src/plugins/condition/plugin.js]: Let's keep the jargon clean. Change 'conditional rules' to 'conditions'

File: src/plugins/condition/pluginSpec.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    describe("shows the conditionSet object is initialized with", () => {
+        beforeEach(() => {
+            openmct.types.get('conditionSet').definition.initialize(mockDomainObject);
+        });
+
+        it('a composition array', () => {
+            expect(Array.isArray(mockDomainObject.composition)).toBeTrue();
+        });
+
+        it('a telemetry object', () => {
```
Comment (shefalijoshi) [inline src/plugins/condition/pluginSpec.js]: Remove this test. I don't think we need the telemetry object for this type yet. We can add it back later if needed.

File: src/plugins/plugins.js [hunk 5]
```diff
@@ -47,6 +47,7 @@ define([
     './goToOriginalAction/plugin',
     './clearData/plugin',
     './webPage/plugin',
+    './condition/plugin',
```
Comment (shefalijoshi) [inline src/plugins/plugins.js]: Rename the folder to "conditionSet"

File: src/plugins/plugins.js [hunk 6]
```diff
@@ -77,6 +78,7 @@ define([
     GoToOriginalAction,
     ClearData,
     WebPagePlugin,
+    ConditionPlugin,
```
Comment (shefalijoshi) [inline src/plugins/plugins.js]: Should be ConditionSetPlugin

File: src/plugins/plugins.js [hunk 7]
```diff
@@ -182,6 +184,7 @@ define([
     plugins.GoToOriginalAction = GoToOriginalAction.default;
     plugins.ClearData = ClearData;
     plugins.WebPage = WebPagePlugin.default;
+    plugins.Condition = ConditionPlugin.default;
```
Comment (shefalijoshi) [inline src/plugins/plugins.js]: Why does this assignment use "default"? 
Should probably just be `plugins.ConditionSet = ConditionSetPlugin`

File: src/plugins/condition/plugin.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2018, United States Government
```
Comment (shefalijoshi) [inline src/plugins/condition/plugin.js]: Change copyright to 2014-2019

File: src/plugins/plugins.js (hunk 7 above)
Comment (JoelMcKinnon) [reply src/plugins/plugins.js]: I based this on WebPage and it just worked. If I remove the ".default" it breaks. Not sure why that's not the case with ClearData and several others. For now I'll leave it as is.

File: src/plugins/condition/plugin.js (hunk 8 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/plugin.js]: BTW - I found 788 hits on the 2018 copyright. Changed it just for the files I'm adding for now. We'll need to do it again in two weeks.

File: src/plugins/plugins.js (hunk 7 above)
Comment (psarram) [reply src/plugins/plugins.js]: @JoelMcKinnon @shefalijoshi Those plugins use ES6 export default and that's why `.default` is needed. The other plugins use ReuireJS for loading modules.

Comment (shefalijoshi) [reply src/plugins/plugins.js]: OK! thanks.

**Comment by shefalijoshi** [comment]
@JoelMcKinnon Approving this, but please add the Author Checklist before you merge it.

**Pull request merged**